create.go

```diff
@@ -91,31 +87,6 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func (r *Resource) applyPatches(ctx context.Context, accessor metav1.Object, patches []Patch) error {
```

Comment:
Everything started with creation. So everything was in this file. Now there is shared code for deletion as well. Thus this gets moved to a more common place.